### PR TITLE
Added XML formatter to Jinja context

### PR DESCRIPTION
**Pull request by blarghmatey**
In order to be able to render properly formatted XML data from an
arbitrarily nested Pillar data structure I added an XML format
function to the jinja.py utility module.

### What does this PR do?
Adds an `xml` formatter to the Jinja context

### What issues does this PR fix or reference?
N/A

### New Behavior
An arbitrarily nested data structure can be passed through the XML filter to render a properly formatted XML document. e.g. `{{ {'foo': {'bar': ['baz', 'quux']}}|xml }}`

### Tests written?

No

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

### CODE REVIEW

File: salt/utils/jinja.py [hunk 1]
```diff
@@ -6,29 +6,31 @@
...
 import logging
+import pipes
+import pprint
 import re
-from os import path
+import uuid
 from functools import wraps
+from os import path
+from xml.dom import minidom
+from xml.etree.ElementTree import Element, SubElement, tostring
```
Comment (mirceaulinic) [inline salt/utils/jinja.py]: Is `xml` a Python standard lib?

Comment (blarghmatey) [inline salt/utils/jinja.py]: Yes it is, for both Python 2 and 3 https://docs.python.org/2/library/xml.html

**Comment by whiteinge** [comment]
Cool addition. I've been wanting an XML serializer in Salt for a while.

**Comment by blarghmatey** [comment]
I found a couple of edge cases that I am addressing which should be ready tomorrow.

**Comment by cachedout** [comment]
@blarghmatey Sounds good. Just keep us posted. :]

**Comment by blarghmatey** [comment]
@cachedout I think it's ready for review now.

**Comment by cachedout** [comment]
@blarghmatey Apologies that it took me so long to get back to this. There is now a merge conflict, unfortunately. Could you please rebase it? Then I'll take a look and we'll hopefully get this in. Thanks!

**Comment by blarghmatey** [comment]
@cachedout it's rebased now. Thanks 😄

**Comment by cachedout** [comment]
Hi @blarghmatey 

We need the following lint warning fixed before we can merge this:

```
************* Module salt.utils.jinja
utils/jinja.py:773: [W1699(incompatible-py3-code), ] Incompatible Python 3 code found. Proposed fix:
                 if isinstance(attrs, dict):
                     sub.attrib = {attr: str(val) for attr, val in attrs.items()
                                   if not isinstance(val, (dict, list))}
-                for tag, val in filter(
-                        lambda item: isinstance(item[1], (dict, list)),
-                        normalize_iter(attrs)):
+                for tag, val in [item for item in normalize_iter(attrs) if isinstance(item[1], (dict, list))]:
                     recurse_tree(((tag, val),), sub)
             return sub
 
utils/jinja.py:175: [W0222(signature-differs), PrintableDict.__repr__] Signature differs from overridden '__repr__' method
utils/jinja.py:239: [C1801(len-as-condition), to_bool] Do not use `len(SEQUENCE)` as condition value

```

**Comment by blarghmatey** [comment]
@cachedout it looks like all of the tests passed. If there is any feedback on the content of the code itself I would be happy to discuss futher.

**Pull request merged**